src/librustdoc/json/conversions.rs

```diff
@@ -170,12 +171,24 @@ impl FromWithTcx<clean::TypeBindingKind> for TypeBindingKind {
     }
 }
 
-crate fn from_def_id(did: FakeDefId) -> Id {
+crate fn from_item_id(did: ItemId) -> Id {
+    struct DisplayDefId(DefId);
+
+    impl fmt::Display for DisplayDefId {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            write!(f, "{}:{}", self.0.krate.as_u32(), u32::from(self.0.index))
+        }
+    }
+
     match did {
-        FakeDefId::Real(did) => Id(format!("{}:{}", did.krate.as_u32(), u32::from(did.index))),
-        // We need to differentiate real and fake ids, because the indices might overlap for fake
-        // and real DefId's, which would cause two different Id's treated as they were the same.
-        FakeDefId::Fake(idx, krate) => Id(format!("F{}:{}", krate.as_u32(), u32::from(idx))),
+        ItemId::DefId(did) => Id(format!("{}", DisplayDefId(did))),
+        ItemId::Blanket { for_, trait_ } => {
```

Comment:
This will cause duplicates if there a multiple blanket impls with different bounds, right?

Reply:
Can you give an example of multiple blanket implementations with different bounds? I couldn't figure it out

Reply:
Oh yes it does. It even panics when trying to document std :sweat_smile:

Reply:
Can you just wrap the impl ID? I'm not sure why this doesn't use DefId in the first place ...

Reply:
Uhm I guess. That's much better yeah